build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/RunParallelTargets.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.BuildTools.PrepTasks.Xamarin.Android.BuildTools.PrepTasks
+{
+	public class RunParallelTargets : Task
+	{
+		[Required]
+		public ITaskItem          ProjectFile                         {get; set;}
+
+		public string             MSBuildBinaryLogParameterPrefix     {get; set;}
+
+		public ITaskItem[]        Targets                             {get; set;}
+
+		public override bool Execute ()
+		{
+			if (Targets?.Length == 0)
+				return true;
+
+			var processes = new Process [Targets.Length];
+			for (int i = 0; i < Targets.Length; ++i) {
+				processes[i]  = CreateProcess (Targets[i].ItemSpec);
+			}
+
+			bool success  = true;
+			for (int i = 0; i < processes.Length; ++i) {
+				processes[i].WaitForExit ();
+				if (processes[i].ExitCode != 0) {
+					Log.LogError ($"Execution of target {Targets [i].ItemSpec} exited with code {processes [i].ExitCode}.");
+					success = false;
+				}
+			}
+			return success;
+		}
+
+		Process CreateProcess (string target)
+		{
+			var binlogOption  = "";
+			if (!string.IsNullOrEmpty (MSBuildBinaryLogParameterPrefix)) {
+				var date  = DateTime.Now.ToString ("yyyyMMddTHHmmss");
+				// -$([System.DateTime]::Now.ToString ("yyyyMMddTHHmmss"))-Target-%(Identity).binlog
+				binlogOption = $"{MSBuildBinaryLogParameterPrefix}-{date}-Target-{target}.binlog\"";
+			}
+			var command     = $"\"{ProjectFile.ItemSpec}\" {binlogOption} /t:{target}";
+			var parameters  = new ProcessStartInfo ("msbuild", command) {
```

Comment:
Ok, I see a problem here now. How do we know this is the _right_ MSBuild? Also, `msbuild` might not be in `%PATH%` on Windows.

I think you can pass `$(MSBuildBinPath)` into this task and combine with `MSBuild.exe` to get the full path.